### ISSUE TITLE
"Timeout was reached on Handle" when trying to sync large PDF file.

### ISSUE THREAD

**Issue by StrikerMan780** [skilion/onedrive#92]
Here's the crash log.

```
std.net.curl.CurlTimeoutException@std/net/curl.d(4039): Timeout was reached on handle DF4E30
----------------
??:? pure @safe void std.exception.bailOut!(std.net.curl.CurlTimeoutException).bailOut(immutable(char)[], ulong, const(char[])) [0x5f8b4d]
??:? pure @safe bool std.exception.enforce!(std.net.curl.CurlTimeoutException, bool).enforce(bool, lazy const(char)[], immutable(char)[], ulong) [0x5f8ac9]
??:? void std.net.curl.Curl._check(int) [0x5d703d]
??:? int std.net.curl.Curl.perform(std.typecons.Flag!("throwOnError").Flag) [0x5d74a0]
??:? int std.net.curl.HTTP.perform(std.typecons.Flag!("throwOnError").Flag) [0x5d425f]
??:? void onedrive.OneDriveApi.download(const(char)[], immutable(char)[]) [0x593777]
??:? void onedrive.OneDriveApi.downloadById(const(char)[], immutable(char)[]) [0x59159c]
??:? void sync.SyncEngine.applyNewItem(itemdb.Item, immutable(char)[]) [0x59cba6]
??:? void sync.SyncEngine.applyDifference(std.json.JSONValue) [0x59c550]
??:? void sync.SyncEngine.applyDifferences() [0x59b6d0]
??:? void main.performSync(sync.SyncEngine) [0x58ba4b]
??:? _Dmain [0x58b28e]
??:? _D2rt6dmain211_d_run_mainUiPPaPUAAaZiZ6runAllMFZ9__lambda1MFZv [0x5a8ade]
??:? void rt.dmain2._d_run_main(int, char**, extern (C) int function(char[][])*).tryExec(scope void delegate()) [0x5a8a28]
??:? void rt.dmain2._d_run_main(int, char**, extern (C) int function(char[][])*).runAll() [0x5a8a9a]
??:? void rt.dmain2._d_run_main(int, char**, extern (C) int function(char[][])*).tryExec(scope void delegate()) [0x5a8a28]
??:? _d_run_main [0x5a8999]
??:? main [0x58ea95]
??:? __libc_start_main [0x60b7982f]
```

This seems to happen when getting large files on a slow connection. Not getting enough time to complete? The connection itself is stable and fine.

**Comment by rodamilans** [comment]
I received the same error. Did anyone resolve this problem?

**Comment by danielasun** [comment]
This seems to happen any time you're downloading files for any period over a couple of hours.

**Comment by skilion** [comment]
Fixed in a3b25c45f52da8a9a7806a3279b726ca49a98577
